lib/awful/permissions/init.lua

```diff
@@ -723,9 +723,14 @@ screen.connect_signal("property::workarea", function(s)
 end)
 
 -- Enable sloppy focus, so that focus follows mouse.
-client.connect_signal("mouse::enter", function(c)
-    c:emit_signal("request::autoactivate", "mouse_enter", {raise=false})
-end)
+if awesome.api_level > 4 then
+    --TODO v5: Remove the code from `rc.lua`. It cannot be done yet because we
+    -- cannot know if the user removed it to disable sloppy focus or because
+    -- they want to use the permissions to manage it.
+    client.connect_signal("mouse::enter", function(c)
```

Comment:
but it seems it will still require changing to smth like

```lua
function permission.sloppy_focus(c)
    c:emit_signal("request::autoactivate", "mouse_enter", {raise=false})
end

client.connect_signal("mouse::enter", permissions.sloppy_focus)
```

Reply:
I don't think this is the best way forward. Disconnecting `request::` signals has caused some bugs report in the past (ie, broke maximization). It is error prone and hard to future-proof. The current way to disable this is

    c:deny("autofocus", "mouse_enter")

Once I am done with the notification PR, the follow-up will add this to `ruled.client`:

```lua
ruled.client.append_rule {
  [...]
  properties = {
    denied_permissions = {
      autofocus = {
        "mouse_enter"
      }
    }
  }
```

The doc will be very explicit about the permissions and the contexts (they are already listed in "show more", but I have some code to auto-generate the index/table_of_content for them). The permissions framework will work on all objects (client, notification, tag, screen, clipboard).

That's my vision for this API going forward. I think it will be great and easier to understand than the current `disconnect_signal` thing.

Reply:
> The current way to disable this is
> `c:deny("autofocus", "mouse_enter")`

did i missed smth or it isn't documented yet?

Reply:
> did i missed smth or it isn't documented yet?

It's there:
https://awesomewm.org/apidoc/core_components/client.html#deny

If you search some pages for "1 permission" and click "show more" on these entries, you will see the table of permissions. There will be much better doc/examples in the PR that adds it to the rules.

Reply:
i mean it would be better to mention the sloppy focus case explicitly, mb in FAQ section, or in release_notes for 5.x

Reply:
also i've just tested it with git master,

```lua
  awful.rules.rules = {
      { rule = { },
        properties = {
          focus = awful.client.focus.filter,
          raise = true,
          keys = awesome_context.clientkeys,
          buttons = awesome_context.clientbuttons,
          placement =
            awful.placement.centered +
            awful.placement.no_overlap +
            awful.placement.no_offscreen,
          size_hints_honor = false,
          screen = awful.screen.preferred,
        },
        callback = function(c)
          c:deny("autofocus", "mouse_enter")
          awful.client.setslave(c)
          apply_delayed_rule(c)
        end
      },
```

adding `c:deny("autofocus", "mouse_enter")` doesn't disable the sloppy focus

Reply:
after a small investigation it seems you meant `c:deny("autoactivate", "mouse_enter")`?

Reply:
>  it seems you meant c:deny("autoactivate", "mouse_enter")?

Yes, sorry about the naming mistake in my previous comment.